server/routes/routes.js

```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const userService = require('../../server/services/user');
+const roleService = require('../../server/services/role');
+const documentService = require('../../server/services/doc');
+const validate = require('../middlewares/auth');
+
+const router = express.Router();
+
+// Setup default route
+router.get('/', (req, res) => {
+  res.status(200).send({
+    message: 'Welcome to the Document Management App!'
+  });
+});
+
+// Route to create and retreive users.
+router.route('/users')
+  .post(userService.createUser)
+  .get(validate.validateAdmin, userService.getAllUsers);
+
+// Route to retrieve single user data
+router.route('/users/:username')
+  .get(validate.validateToken, userService.getUser)
+  .put(validate.validateToken, userService.updateUser)
+  .delete(validate.validateAdmin, userService.deleteUser);
+
+// User login route
+router.route('/login')
+  .get(userService.login);
+
+router.route('/logout')
+  .get(userService.logout);
+
+// Route to create roles
+router.route('/roles')
+  .post(validate.validateAdmin, roleService.createRole);
+
+// Retrieve all roles
+router.route('/roles/all')
+  .get(validate.validateAdmin, roleService.getAllRoles);
+
+// Retrieve a single role
+router.route('/roles/:title')
+  .get(validate.validateAdmin, roleService.getRole);
+
+// Route for documents
+router.route('/documents')
+  .post(validate.validateToken, documentService.createDocument)
+  .get(validate.validateToken, documentService.getAllDocuments);
+
+// Retrieve a single document
+router.route('/documents/:id')
+  .get(validate.validateToken, documentService.getDocument);
+
+// Retrieve limited documents
+router.route('/documents/all/:limit')
+  .get(validate.validateToken, documentService.getAllDocuments);
+
+// Retrieve limited documents with offset
+router.route('/documents/all/:offset/:limit')
```

Comment:
The `offset` and `limit` should be query not params. The url should look like `/documents/all?offset=5&limit=10` not `/documents/all/5/10`